### PR TITLE
Adds hot phase progress copy

### PR DESCRIPTION
**Pull request by itsjoekent**
Relevant issue https://github.com/DoSomething/dosomething/issues/4446 but it does not fix. Issue will be re-assigned to @sbsmith86 after this is merged

Adds progress copy settings with variable support needed and calculates reportback progress. Also adds some general hot module code.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.admin.inc [hunk 1]
```diff
@@ -283,6 +283,48 @@ function dosomething_campaign_admin_config_form($form, &$form_state) {
...
+    '#default_value' => variable_get('dosomething_campaign_81_99_progress_copy', t('We’re almost there! Keep showing off your work. You can add photos and update your submission throughout the campaign.')),
+    '#description' => t('Sets the hot module progress copy between 81 and 99%'),
+  ];
+
+  $form['goals']['progress_copy']['dosomething_campaign_100_progress_copy'] = [
+    '#type' => 'textarea',
+    '#title' => t('Hot module 100% progress copy'),
+    '#default_value' => variable_get('dosomething_campaign_100_progress_copy', t('Woohoo! Congratulations everyone for crushing our goal of [campaign_goal] [node:field_reportback_noun] [node:field_reportback_verb]! Can you help push us even higher!? Keep submitting photos of your work. You can update your submission with new photos/info at any time.')),
+    '#description' => t('Sets the hot module progress copy for 100%'),
+  ];
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.admin.inc:326]: can we for loop this stuffs?

Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.admin.inc:326]: I guess it's the same, if we have to have an array of values, maybe this is better

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.admin.inc:326]: We _could_ but we'd still need to individually set the default values.

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc [hunk 2]
```diff
@@ -99,7 +104,38 @@ function dosomething_campaign_preprocess_common_vars(&$vars, &$wrapper) {
...
+/**
+ * Preprocess variables for hot module.
+ */
+function dosomething_campaign_preprocess_hot_module(&$vars, &$wrapper) {
+  $node = node_load($vars['nid']);
+  $goal = dosomething_helpers_get_variable('node', $vars['nid'], 'goal');
+  $rb_progress = dosomething_reportback_get_reportback_total_by_nid($node->nid);
+  $rb_percent = ($rb_progress / $goal) * 100;
+
+  if($rb_percent <= 20) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc:122]: need moar :space_invader:  `if ($rb_percent) {`

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc:122]: should we assign a default value of 0 instead so there is always a progress copy?

Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc:122]: :+1:

**Comment by angaither** [comment]
@deadlybutter the `if` statements all need extra spaceeee

**Comment by itsjoekent** [comment]
oh whoops. ill make another quick PR and fix that

**Comment by angaither** [comment]
@deadlybutter actually I have a PR based on this, I can do it in that one!

**Comment by itsjoekent** [comment]
works for me, thanks

**Pull request merged**